### PR TITLE
Frontend: Remove support for `-experimental-spi-imports` flag

### PR DESCRIPTION
**Pull request by tshortli**
Its functionality has been superseded by `@_spiOnly import`. There are no longer any known clients and the flag was already unsupported in Swift 6, so the functionality is now removed (but the flag is only deprecated for Swift 5).
    
Resolves rdar://136867210.

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test Windows

File: lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -1193,6 +1191,16 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
...
 
   Opts.EnableSPIOnlyImports = Args.hasArg(OPT_experimental_spi_only_imports);
+  if (Args.hasArg(OPT_experimental_spi_imports)) {
+    if (Opts.EffectiveLanguageVersion.isVersionAtLeast(6)) {
+      Diags.diagnose(SourceLoc(), diag::flag_unsuppored,
+                     "-experimental-spi-only-imports");
+      HadError = true;
+    } else {
+      Diags.diagnose(SourceLoc(), diag::warn_flag_deprecated,
+                     "-experimental-spi-only-imports");
```
Comment (xymus) [inline lib/Frontend/CompilerInvocation.cpp:1201]: The two diagnostics should refer to `-experimental-spi-imports` right?

Comment (tshortli) [reply lib/Frontend/CompilerInvocation.cpp:1201]: Oops yes

**Review by xymus — APPROVED** [review]
Thank you for clearing this old hack!

File: lib/Frontend/CompilerInvocation.cpp (hunk 1 above)
Comment (tshortli) [reply lib/Frontend/CompilerInvocation.cpp]: Fixed

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Pull request merged**